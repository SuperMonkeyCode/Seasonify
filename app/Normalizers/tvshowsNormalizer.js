function tvshowsNormalizer(tvshows) {
  return tvshows.reduce(
    (array, suggestion) =>
      array.concat({
        allocine: suggestion.code,
        name: suggestion.originalTitle,
        localizedName: 'title' in suggestion ? suggestion.title : undefined,
        poster: 'poster' in suggestion ? suggestion.poster.href : undefined,
        year: suggestion.yearStart,
        seasons: {},
        seasonsCount: 0,
      }),
    [],
  );
}

export default tvshowsNormalizer;
