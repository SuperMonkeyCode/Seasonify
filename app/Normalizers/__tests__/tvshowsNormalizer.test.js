import tvshowsNormalizer from '../tvshowsNormalizer';

const input = require('../../Fixtures/apiSearchHappy.json');

test('tvshowsNormalizer output should match snapshot', () => {
  expect(tvshowsNormalizer(input)).toMatchSnapshot();
});
