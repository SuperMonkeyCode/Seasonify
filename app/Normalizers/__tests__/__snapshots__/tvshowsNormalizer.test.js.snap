// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tvshowsNormalizer output should match snapshot 1`] = `
Array [
  Object {
    "allocine": 8090,
    "localizedName": undefined,
    "name": "Happy Endings",
    "poster": "http://fr.web.img2.acsta.net/medias/nmedia/18/78/21/06/20278341.jpg",
    "seasons": Object {},
    "year": 2011,
  },
  Object {
    "allocine": 720,
    "localizedName": "Happy Days - Les jours heureux",
    "name": "Happy Days",
    "poster": "http://fr.web.img6.acsta.net/medias/nmedia/18/36/37/63/18478173.jpg",
    "seasons": Object {},
    "year": 1974,
  },
  Object {
    "allocine": 4876,
    "localizedName": undefined,
    "name": "Happy Town",
    "poster": "http://fr.web.img4.acsta.net/medias/nmedia/18/73/99/08/19237328.jpg",
    "seasons": Object {},
    "year": 2010,
  },
  Object {
    "allocine": 7391,
    "localizedName": undefined,
    "name": "Happy Tree Friends",
    "poster": "http://fr.web.img5.acsta.net/medias/nmedia/18/72/22/91/19157134.jpg",
    "seasons": Object {},
    "year": 2000,
  },
  Object {
    "allocine": 6417,
    "localizedName": undefined,
    "name": "Happy Lucky Bikkuriman",
    "poster": undefined,
    "seasons": Object {},
    "year": 2006,
  },
]
`;
